### PR TITLE
MC-1339 mc-sgx-epid #in-review

### PR DESCRIPTION
**Pull request by jcape**
This implements the rust wrapper around those bits of the SGX API implemented by `libsgx_epid`. It accomplishes this by monkey-patching additional functionality onto the `TargetInfo`, `Quote`, `PlatformInfo`, and `EpidGroupId`.

This will (eventually) supersede the analogous EPID functionality in `mc-attest-trusted`.

### CODE REVIEW

File: sgx/core-types/src/error.rs [hunk 1]
```diff
@@ -3,7 +3,7 @@
 //! SGX Error types
 
 use core::{convert::TryFrom, result::Result as StdResult};
-use failure::Fail;
+use displaydoc::Display;
```
Comment (mfaulk) [inline sgx/core-types/src/error.rs:6]: Cool

File: sgx/epid/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2018-2020 MobileCoin Inc.
+
+//! Traits and support for EPID-based remote attestation.
+
+#![deny(missing_docs)]
```
Comment (mfaulk) [inline sgx/epid/src/lib.rs:5]: Thanks

File: sgx/epid/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+                quote_size,
+            )
+        }
+        .into_result((quote, qe_report))
+    }
+}
+
+bitflags! {
+    /// A set of bitflags describing what kind of reprovisioning to perform, if any.
+    pub struct EpidUpdateConfig: u32 {
```
Comment (mfaulk) [inline sgx/epid/src/lib.rs:117]: I notice that EpidUpdateStatus starts with 1, while this starts with a shift (1 << 1). Just checking if that's intentional.

Comment (jcape) [reply sgx/epid/src/lib.rs:117]: Yeah, bit zero is reserved in the config u32 in the Intel API.

**Pull request merged**